apps/src/templates/projects/submitProjectDialog/SubmitProjectDialog.tsx

```diff
@@ -0,0 +1,98 @@
+import React, {useCallback, useState} from 'react';
+
+import {showShareDialog} from '@cdo/apps/code-studio/components/shareDialogRedux';
+import Button from '@cdo/apps/componentLibrary/button/Button';
+import Link from '@cdo/apps/componentLibrary/link/Link';
+import {BodyTwoText, Heading3} from '@cdo/apps/componentLibrary/typography';
+import AccessibleDialog from '@cdo/apps/sharedComponents/AccessibleDialog';
+import {useAppDispatch, useAppSelector} from '@cdo/apps/util/reduxHooks';
+import i18n from '@cdo/locale';
+
+import {submitProject} from './submitProjectApi';
+import {setShowSubmitProjectDialog} from './submitProjectRedux';
+
+import moduleStyles from './submit-project-dialog.module.scss';
+
+/**
+ * Renders a modal that allows a user to submit a project to be considered for the
+ * featured project gallery.
+ */
+
+export interface SubmitProjectDialogProps {
+  onClose: () => void;
+}
+
+const SubmitProjectDialog: React.FunctionComponent<
+  SubmitProjectDialogProps
+> = ({onClose}) => {
+  const [projectDescription, setProjectDescription] = useState<string>('');
+  const channelId = useAppSelector(state => state.lab.channel?.id);
+  const projectType = useAppSelector(
+    state => state.lab.channel?.projectType
+  ) as string;
+  const dispatch = useAppDispatch();
+
+  const onSubmit = async () => {
+    try {
+      if (channelId && projectType) {
+        submitProject(projectDescription);
+      }
+    } catch (error) {
+      console.error('Publish failed', error);
+    }
+  };
+
+  const onGoBack = useCallback(() => {
```

Comment:
Might be better to pass in a callback for go back so that the parent dialog can decide how to handle it. That would make this easier to eventually add to the non-lab2 share dialog as well.